specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,577 @@
+  HRESULT remove_Stopped(
+      [in] EventRegistrationToken token);
+  /// Creates TextureBuffer that will be referenced by the host and the browser.
+  /// By using the TextureBuffer mechanism, the host does not have to
+  /// send respective texture to the renderer, instead it notifies it
+  /// with internal TextureBuffer id, which is the identity of the TextureBuffer.
+  /// The TextureBuffer is 2D texture, IDXGIResource, format and will be
+  /// exposed through shared HANDLE or IUnknown type through ICoreWebView2StagingTexture.
+  /// Whenever the host has new texture to write, it should ask
+  /// reusable ICoreWebView2StagingTexture from the GetAvailableBuffer,
+  /// which returns ICoreWebView2StagingTexture.
+  /// If the GetAvailableBuffer returns an error, then the host calls the
+  /// CreateBuffer to allocate new TextureBuffer.
+  /// The API also registers created shared handle to the browser once it
+  /// created the resource.
+
+  /// Unit for width and height is texture unit (in texels).
+  /// 'https://learn.microsoft.com/en-us/windows/win32/api/d3d11ns-d3d11-d3d11_texture2d_desc'
+  HRESULT CreateBuffer(
+    [in] UINT32 widthInTexels,
+    [in] UINT32 heightInTexels,
+    [out, retval] ICoreWebView2StagingTexture** buffer);
+  /// Returns reuseable TextureBuffer for video frame rendering.
+  /// Once the renderer finishes rendering of TextureBuffer's video frame, which
+  /// was requested by Present, the renderer informs the host so that it can
+  /// be reused. The host has to create new TextureBuffer with CreateBuffer
+  /// if the API return an error HRESULT_FROM_WIN32(ERROR_NO_MORE_ITEMS).
+  HRESULT GetAvailableBuffer([out, retval] ICoreWebView2StagingTexture** buffer);
+  /// Removes TextureBuffer when the host removes the backed 2D texture.
+  /// The host can save the resources by deleting 2D textures
+  /// when it changes the frame sizes. The API will send a message
+  /// to the browser where it will remove TextureBuffer.
+  HRESULT CloseBuffer([in] ICoreWebView2StagingTexture* buffer);
+  /// Sets rendering image/resource through ICoreWebView2StagingTexture.
+  /// The TextureBuffer must be retrieved from the GetAvailableBuffer or
+  /// created via CreateBuffer.
+  /// It is expected that hhe host writes new image/resource to the local
+  /// shared 2D texture of the TextureBuffer (handle/resource).
+
+  /// `timestampInMicroseconds` is video capture time with microseconds units.
+  /// The value does not have to be exact captured time, but it should be
+  /// increasing order because renderer (composition) ignores incoming
+  /// video frame (texture) if its timestampInMicroseconds is equal or prior to
+  /// the current compositing video frame. It also will be exposed to the
+  /// JS with `VideoFrame::timestamp`.
+  /// (https://docs.w3cub.com/dom/videoframe/timestamp.html).
+  HRESULT PresentBuffer([in] ICoreWebView2StagingTexture* buffer,
+    [in] UINT64 timestampInMicroseconds);

```

Comment:
Let's move the timestamp to the texture interface as a set/get property. The timestamp goes with the visual that's written to the texture so it makes sense that they are written together into one containing object.